src/vm/linux/types.rs

```diff
@@ -1 +1,93 @@
 // SPDX-License-Identifier: Apache-2.0
+
+pub const NR_CPUID_CONFIGS: usize = 12;
+
+/// CPUID_CONFIG is designed to enumerate how the host VMM may configure the
+/// virtualization done by the Intel TDX module for a single CPUID leaf and
+/// sub-leaf. This is equivalent to `struct kvm_tdx_cpuid_config` in the kernel.
+#[derive(Debug, Default, Clone, Copy)]
+#[repr(C)]
+pub struct CpuidConfig {
+    /// EAX input value to CPUID
+    pub leaf: u32,
+    /// ECX input value to CPUID. A value of -1 indicates a CPUID leaf with
+    /// no sub-leaves.
+    pub sub_leaf: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EAX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub eax: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EBX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub ebx: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in ECX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub ecx: u32,
+    /// Enumeration of the configurable virtualization of the value returned
+    /// by CPUID in EDX: a value of 1 in any of the bits indicates that the
+    /// host VMM is allowed to configure that bit.
+    pub edx: u32,
+}
+
+/// Provides information about the Intel TDX module. This is equivalent to
+/// `struct kvm_tdx_capabilities` in the kernel.
+#[derive(Debug)]
+#[repr(C)]
+pub struct Capabilities {
+    /// Bitmap where if any certain bit is 0, it must be 0 in any TD's
+    /// ATTRIBUTES, which specifies various guest TD attributes. The value of
+    /// this field reflects the Intel TDX module capabilities and configuration
+    /// and CPU capabilities.
+    pub attrs_fixed0: u64,
+    /// Bitmap where if any certain bit is 1, it must be 1 in any TD's
+    /// ATTRIBUTES, which specifies various guest TD attributes. The value of
+    /// this field reflects the Intel TDX module capabilities and configuration
+    /// and CPU capabilities.
+    pub attrs_fixed1: u64,
+    /// Bitmap where if any certain bit is 0, it must be 0 in any TD's XFAM.
+    /// XFAM (eXtended Features Available Mask) determines the set of extended
+    /// features available for use by the guest TD.
+    pub xfam_fixed0: u64,
+    /// Bitmap where if any certain bit is 1, it must be 1 in any TD's XFAM.
+    /// XFAM (eXtended Features Available Mask) determines the set of extended
+    /// features available for use by the guest TD.
+    pub xfam_fixed1: u64,
+    /// Supported Guest Physical Address Width
+    pub supported_gpaw: u32,
+    pub padding: u32,
+    /// Set to 0
+    pub reserved: [u64; 251],
```

Comment:
Set `padding` and `reserved` to private to remove the possibility that they are used. Update the comments to reflect that.

```suggestion
    /// Padding space. Ignored.
    _padding: u32,
    /// Reserved space. Ignored.
    _reserved: [u64; 251],
```